submissions/AQuaZar/port-sniffer/sniffer.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import print_function  # Compatibility with python2
+import sys
+import argparse
+import socket
+
+
+def range_type(astr, min=0, max=65535):
+    value = int(astr)
+    if min <= value <= max:
+        return value
+    else:
+        raise argparse.ArgumentTypeError(
+            'value not in range {min}-{max}'.format(min, max))
+
+
+parser = argparse.ArgumentParser(
+    description='Output a list of opened TCP ports on specified host. By default all ports in range (0..65535) will be checked')
+
+parser.add_argument('--host',
+                    metavar='Host',
+                    type=str,
+                    help='IPv4 address or domain name, which ports to check e.g. www.google.com, 192.168.0.1, e.t.c ',
+                    required=True)
+parser.add_argument('--ports',
+                    metavar=('First', 'Last'),
+                    type=range_type,
+                    nargs=2,
+                    default=[0, 65535],
+                    help='To check specific range of ports, define first and last port from range, e.g. --ports 69 420')
+
+args = parser.parse_args()
+
+
+def port_scan(host, first, last):
+    print('Searching')
+    open_ports = []
+    for port in range(first, last+1):
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(0.3)
+        result = sock.connect_ex((host, port))
+
+        if result == 0:
+            open_ports.append(port)
+            print('.', end='')
+            sys.stdout.flush()
+        sock.close()
+    print('\n')
+    return open_ports
+
+
+if __name__ == '__main__':
+    ports = []
+    try:
+        ports = port_scan(args.host, args.ports[0], args.ports[1])
```

Comment:
`port_scan` function already returns `[]` in case of no open ports
so you don't need to define `ports` on line 52

Reply:
I tried to prevent situation when user interrupts the program with Ctrl+C during search and so it calls undefined list, was that a good idea? 

![image](https://user-images.githubusercontent.com/36030534/66311206-9f240100-e916-11e9-9f00-4ee5bac8ea03.png)

Reply:
sure